packages/fast-components-react-msft/src/text-action/text-action.spec.tsx

```diff
@@ -0,0 +1,323 @@
+                .find(`.${managedClasses.textAction}`)
+                .first()
+                .childAt(0)
+                .type()
+        ).toEqual("svg");
+    });
+
+    test("should render a glyph in the after position if passed as a prop", () => {
+        const handledProps: TextActionHandledProps = {
+            managedClasses,
+            afterGlyph: (classname?: string): React.ReactNode => {
+                return (
+                    <svg
+                        width="16"
+                        height="16"
+                        viewBox="0 0 16 16"
+                        fill="none"
+                        xmlns="http://www.w3.org/2000/svg"
+                        className={classname}
+                    >
+                        <path d="M10.3906 9.39844C11.099 9.64323" fill="black" />
+                    </svg>
+                );
+            },
+        };
+
+        const props: TextActionProps = { ...handledProps };
+        const rendered: any = mount(<TextAction {...props} />);
+
+        expect(rendered.find("svg")).not.toBe(undefined);
+        expect(
+            rendered
+                .find(`.${managedClasses.textAction}`)
+                .first()
+                .childAt(1)
+                .type()
+        ).toEqual("svg");
+    });
+
+    test("should render a button only if a glyph and button are passed in the same (after) position", () => {
+        const handledProps: TextActionHandledProps = {
+            managedClasses,
+            button: (classname?: string, disabled?: boolean): React.ReactNode => {
+                return (
+                    <Button type={"submit"} className={classname} disabled={disabled}>
+                        {"Text action"}
+                    </Button>
+                );
+            },
+            afterGlyph: (classname?: string): React.ReactNode => {
+                return (
+                    <svg
+                        width="16"
+                        height="16"
+                        viewBox="0 0 16 16"
+                        fill="none"
+                        xmlns="http://www.w3.org/2000/svg"
+                        className={classname}
+                    >
+                        <path d="M10.3906 9.39844C11.099 9.64323" fill="black" />
+                    </svg>
+                );
+            },
+        };
+
+        const props: TextActionProps = { ...handledProps };
+        const rendered: any = mount(<TextAction {...props} />);
+
+        expect(
+            rendered
+                .find(`.${managedClasses.textAction}`)
+                .find("svg")
+                .exists()
+        ).toBe(false);
+        expect(
+            rendered
+                .find(`.${managedClasses.textAction}`)
+                .find("button")
+                .exists()
+        ).toBe(true);
+    });
+
+    test("should render a button only if a glyph and button are passed in the same (before) position", () => {
```

Comment:
nit: I read this as a button should be rendered only in the case that both a glyph and button are passed. Consider `should render only a button if...`?